src/thor/costmatrix.cc

```diff
@@ -65,24 +65,44 @@ float CostMatrix::GetCostThreshold(const float max_matrix_distance) {
 
 // Clear the temporary information generated during time + distance matrix
 // construction.
-void CostMatrix::Clear() {
+void CostMatrix::clear() {
```

Comment:
this tries to reclaim all allocations. that might not be great for a long running server though. some of these containers could stand to keep some heap space so that they dont have to immediately reallocate on the next request. should we make at least some default values other than `0` to shrink to so that we dont have to always pay the reallocation penalty? other algorithms have configuration (this could too) which controls what these maximum-allocation-after-request numbers, that way mobile applications can set it to `0` to get all their ram back and servers can use some reasonable numbers to avoid the realloc penalty. @dnesbitt61 what do you think?

Reply:
The problem with CostMatrix is that for an m sources by n targets request there are m+n EdgeLabel vectors, m+n EdgeStatus objects, and m+n adjacency lists. I clear, resize, and shrink these vectors so the internal container memory is deallocated. Which allocated memory would you retain? 
Contrast that to TimeDistanceMatrix which has one of each of these objects, so retaining some allocation (for EdgeLabels for example) makes sense.

Reply:
yep good point about the nesting. i guess maybe only that of the top level containers could be regulated via config so we at least avoid some level of reallocation? honestly i havent measured the throughput difference so maybe for now we can do something like this, give it a test, and make it configureable in a future pr?